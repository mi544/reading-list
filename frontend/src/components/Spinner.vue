<template>
  <div class="loading-spinner">
    <div class="spinner"></div>
    <div class="spinner"></div>
    <div class="spinner"></div>
  </div>
</template>

<script>
export default {
  name: 'Spinner',
}
</script>

<style lang="scss">
@keyframes loading-spinner {
  0% {
    transform: rotate(305deg);
  }
  50% {
    transform: rotate(60deg);
  }
  100% {
    transform: rotate(305deg);
  }
}

.loading-spinner {
  display: flex;
  width: 100px;
  height: 35px;
  justify-content: space-between;
  align-items: center;

  & > .spinner {
    width: 20px;
    height: 20px;
    background-color: #ffffff;
    border-radius: 4px;
  }

  $spinners: ((1, 1s, -0.2s) (2, 1.7s, 0s) (3, 1s, -0.2s));

  @each $num, $duration, $delay in $spinners {
    & > div:nth-child(#{$num}) {
      animation: loading-spinner
        $duration
        $delay
        infinite
        cubic-bezier(0.65, -0.05, 0.38, 0.96);
    }
  }
}
</style>
