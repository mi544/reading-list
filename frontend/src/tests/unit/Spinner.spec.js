/* eslint-disable no-undef */
import { shallowMount } from '@vue/test-utils'
import Spinner from '../../components/Spinner.vue'

describe('Spinner', () => {
  it('mounts successfully', () => {
    const wrapper = shallowMount(Spinner)
    expect(wrapper.exists()).toBeTruthy()
  })
  it('matches snapshot', () => {
    const wrapper = shallowMount(Spinner)
    expect(wrapper.element).toMatchSnapshot()
  })
})
