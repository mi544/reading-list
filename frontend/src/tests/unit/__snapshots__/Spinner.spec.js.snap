// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner matches snapshot 1`] = `
<div
  class="loading-spinner"
>
  <div
    class="spinner"
  />
  <div
    class="spinner"
  />
  <div
    class="spinner"
  />
</div>
`;
